server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -193,49 +206,106 @@ private MergeResult partialReduce(MergeTask task,
             processedShards.add(new SearchShard(target.getClusterAlias(), target.getShardId()));
         }
         progressListener.notifyPartialReduce(processedShards, topDocsStats.getTotalHits(), newAggs, numReducePhases);
-        return new MergeResult(processedShards, newTopDocs, newAggs);
+        return new MergeResult(processedShards, newTopDocs, newAggs, hasAggs ? newAggs.getBinarySize() : 0);
```

Comment:
Its probably worth commenting about how we leave the results un-serialized but we send their serialized size because serializing is slow.

Reply:
https://github.com/elastic/elasticsearch/pull/62223/commits/7178dab4566971b0b0daee19f2b22a8f597f61b1